src/algokit/cli/generate.py

```diff
@@ -3,12 +3,36 @@
 
 import click
 
+from algokit.core.generate import load_custom_generate_commands
 from algokit.core.typed_client_generation import ClientGenerator
 
 logger = logging.getLogger(__name__)
 
 
-@click.group("generate")
+class GeneratorGroup(click.Group):
+    def get_command(self, ctx: click.Context, cmd_name: str) -> click.Command | None:
+        return_value = super().get_command(ctx, cmd_name)
+
+        if return_value is not None:
+            return return_value
+
+        commands = load_custom_generate_commands(Path.cwd())
+
+        for command in commands:
+            if command.name == cmd_name:
+                return command
+
+        return None
+
+    def list_commands(self, ctx: click.Context) -> list[str]:
+        predefined_command_names = super().list_commands(ctx)
+        dynamic_commands = load_custom_generate_commands(Path.cwd())
+        dynamic_command_names = [command.name for command in dynamic_commands if command and command.name]
```

Comment:
similarly, this could then just be `dynamic_command_names = list(dynamic_commands)`

Reply:
@achidlow done 👍